.github/workflows/deploy.yml

```diff
@@ -93,3 +87,136 @@ jobs:
           SLACK_COLOR: ${{job.status}}
           SLACK_FOOTER: ""
           MSG_MINIMAL: actions url,commit,ref
+  cypress-test:
+    name: Cypress Tests
+    needs: deploy
+    if: ${{ github.ref != 'refs/heads/prod' }}
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        containers: [test.spec.js, test2.spec.js]
+    steps:
+      - name: set branch_name
```

Comment:
Since this `cypress-test` job depends on `deploy` already, are lines 100 to 144 needed? They are mostly repeats of what is already done before deploy.  I'm not certain since you are launching matrix jobs. Do you know?

Reply:
Yes, to set up values and uses the actions specific to it as well.